descopesdk/src/main/java/com/descope/internal/http/HttpClient.kt

```diff
@@ -71,12 +82,25 @@ internal open class HttpClient(
             val responseCode = connection.responseCode
             if (responseCode == HttpsURLConnection.HTTP_OK) {
                 val response = connection.inputStream.bufferedReader().use { it.readText() }
+                logger?.log(Debug, "Received response body", response)
                 decoder(response)
             } else {
-                // TODO: handle error responses and network errors
-                throw Exception("Network error")
+                val response = connection.errorStream.bufferedReader().use { it.readText() }
+                exceptionFromResponse(response)?.run {
+                    logger?.log(Info, "Network call failed with server error", url, responseCode, this)
+                    throw this
+                }
+                logger?.log(Info, "Network call failed with server error", url, responseCode)
+                throw exceptionFromResponseCode(responseCode) ?: Exception("Network error")
+            }
+        } catch (e: Exception) {
+            if (e !is DescopeException) {
+                logger?.log(Error, "Network call failed with network error", url, e)
```

Comment:
Is this a new language feature where you don't need to write the enum class (ie, `DescopeLogger.Level.Error`)?

```kotlin
logger?.log(Error, "Network call failed with network error", url, e)
```

Reply:
I wish. There's a static import up top